python/cudf/cudf/utils/ioutils.py

```diff
@@ -2231,3 +2238,28 @@ def _prefetch_remote_buffers(
 
     else:
         return paths
+
+
+def _add_df_col_struct_names(
+    df: cudf.DataFrame, child_names_dict: dict
+) -> None:
+    for name, child_names in child_names_dict.items():
+        col = df._data[name]
+        df._data[name] = _update_col_struct_field_names(col, child_names)
```

Comment:
Just thinking out loud based on @wence- review. We probably don't want to eagerly copy, so maybe we can optionally copy if the column and children share data?